boards/stm32mp157c-dk2/Makefile.include

```diff
@@ -0,0 +1,20 @@
+# we use shared STM32 configuration snippets
+INCLUDES += -I$(RIOTBOARD)/common/stm32/include
+
+# set default port depending on operating system
+PORT_LINUX ?= /dev/ttyACM0
+PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.SLAB_USBtoUART*)))
+
+# setup serial terminal
+include $(RIOTMAKE)/tools/serial.inc.mk
+
+# this board has an on-board ST-link adapter
+OPENOCD_CORE = "stm32mp15x.cm4"
+OPENOCD_CONFIG = board/stm32mp15x_dk2.cfg
```

Comment:
Please do as done for other stm32 boards and add a `stm32mp1.cfg` in `boards/common/stm32/dist`, so something like

```
source [find target/stm32mp15x.cfg]
reset_config srst_only
```

and add for the BOARD.

```
OPENOCD_ADAPTER_INIT ?= \
  -c 'source [find interface/stlink-dap.cfg]' \
  -c 'transport select dapdirect_swd'
```

Reply:
Please can you explain me more ?
As the board is already mainlined into openocd, I do not see why I should rewrite it this way.
https://sourceforge.net/p/openocd/code/ci/master/tree/tcl/board/stm32mp15x_dk2.cfg

Reply:
@fjmolinas @aabadie
If it is in order to homogenize the way of doing things, I understand and I can do it.
Lack of tags/version on openocd does not make things easy. It makes me think that adding the openocd commit to use in doc.txt of the board could be a good thing. What do you think about it ?

Reply:
Sorry for taking so long to respond, this got lost in my mailbox. I would rather have a common configuration because:

- it can be re-used by other boards
- some of the `tcl/board` configuration perform some initialization that can conflict with the other we perform all the `do_%` commands. I can't recall right now an example (maybe `OPENOCD_RESET_USE_CONNECT_ASSERT_SRST` is one of those cases?).
- as you say to keep things homogenuous

> It makes me think that adding the openocd commit to use in doc.txt of the board could be a good thing. What do you think about it ?

Yes if a specific commit is needed it would be nice to have it in the doc I think thats how @aabadie has done it for most cases.

Reply:
no worries.
Ack for changes, thank you.